.github/workflows/Supposed_test.yml

```diff
@@ -0,0 +1,42 @@
+name: my_try_in_CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the new branch
+  push:
+    branches: [ Br_1 ]
+  pull_request:
+    branches: [ Br_1 ]
```

Comment:
What is interesting here: when exactly the workflow will run:

1. On the `Br_1` branch (not any development branch).
2. On a pull request that targets the `Br_1` branch (not the default branch).

There are different approaches that works good in different development flows. However usually you want to run testing on push to every branch (disregarding whether you're in the main repository or in a fork):

```yaml
on:
  push:
```

And usually you want to run testing on every pull request:

```yaml
on:
  pull_request:
```

It may be written in the short way:

```yaml
on: [push, pull_request]
```

There is one downside in this approach: when you open a pull request from a branch within the repository, the testing will run twice. It is possible to [work it around](https://github.com/h2o/h2o/blob/08f0fcad7f423df285c5b886e74b2e1efe0c38e0/.github/workflows/ci.yml#L10-L12), though.